### DISCUSSION TITLE
.env inside AL code.

### DISCUSSION THREAD

**Discussion by joshuahiwat** [microsoft/alguidelines#221] [General]
Hi All,

I'm a php developer and currently learning a lot of AL code.

I want to keep my code clean and readable for others who are working on it in the future. Some of those things are the environment variables. Why is there no option to create a kind of .env file or something where you can place your important information, like API keys or slugs. 

I know there is a isolated storage, but that is not what I mean. 

Any ideas or did I take a look at the wrong place?

**Comment by TheDoubleH** [comment]
I would create a Setup table, where those values are stored. For Security purposes I would NEVER store authentication details in a repository.

**Reply by joshuahiwat** [reply]
> I would create a Setup table, where those values are stored. For Security purposes I would NEVER store authentication details in a repository.

That isn't a great solution to store environment details. Within several program languages, you'll store those inside a .env file and you'll share the .env.example file within your repository. Why is there no way to have this within AL?

**Reply by christianbraeunlich** [reply]
> That isn't a great solution to store environment details. Within several program languages, you'll store those inside a .env file and you'll share the .env.example file within your repository. Why is there no way to have this within AL?

Hi @joshuahiwat, also using the .env file in other (non-AL) projects. Would you like to reference this question in a new issue in the official AL repo? (https://github.com/microsoft/AL)
btw: there would also be the possibility to use the AzureKeyVault
